Mathlib/Topology/UnitInterval.lean

```diff
@@ -172,6 +213,25 @@ theorem mul_pos_mem_iff {a t : ℝ} (ha : 0 < a) : a * t ∈ I ↔ t ∈ Set.Icc
 theorem two_mul_sub_one_mem_iff {t : ℝ} : 2 * t - 1 ∈ I ↔ t ∈ Set.Icc (1 / 2 : ℝ) 1 := by
   constructor <;> rintro ⟨h₁, h₂⟩ <;> constructor <;> linarith
 
+instance : LinearOrderedCommMonoidWithZero I where
+  zero_mul i := zero_mul i
+  mul_zero i := mul_zero i
+  zero_le_one := nonneg'
+  mul_le_mul_left i j h_ij k := by
+    simp only [← Subtype.coe_le_coe, coe_mul]
+    apply mul_le_mul le_rfl ?_ (nonneg i) (nonneg k)
+    simp [h_ij]
+  le_total := LinearOrder.le_total
+  decidableLE := LinearOrder.decidableLE
+  min_def := LinearOrder.min_def
+  max_def := LinearOrder.max_def
+  compare_eq_compareOfLessAndEq i j := by
+    simp only [compare, compareOfLessAndEq, Subtype.coe_lt_coe]
+    split_ifs
+    all_goals try rfl
+    case pos h_ne h_eq => exact h_ne <| Subtype.coe_inj.mpr h_eq
+    case neg h_eq h_ne => exact h_ne <| Subtype.coe_inj.mp h_eq
```

Comment:
You should be able to just not specify this

Reply:
I should, and I assumed this as well.
However, leaving `compare_eq_compareOfLessAndEq` away leaves me with the unsolved case `mk.mk` with goal `compare ⟨val✝¹, property✝¹⟩ ⟨val✝, property✝⟩ = compareOfLessAndEq ⟨val✝¹, property✝¹⟩ ⟨val✝, property✝⟩`.